### ISSUE TITLE
Interface name checks are too strict, esp. for kind: ext-container but also crpd

### ISSUE THREAD

**Issue by mgreve** [srl-labs/containerlab#1275]
The interface name checks (added around v0.35) are too strict and are causing our labs to fail deployment. For instance, for the **kind: ext-container**, the default check applies, which means interfaces must be named ethX. Such a check is unreasonably strict given that it's an ext-container kind (which could be anything).

It is also causing trouble for the **crpd** kind in our labs, where our interface names are etX. The default check also applies here since the crpd kind does not override the check.

Here's an example lab showing the problem:

```
name: tmplab

mgmt:
  network: tmp

topology:
  nodes:
    container1:
      kind: ext-container
    container2:
      kind: ext-container

  links:
    - endpoints: ["container1:swp11", "container2:et12"]
```

Deploying such a lab using v0.34 works fine, but fails using v0.35, because interfaces are not named ethX (the default interface name check). 

**Would it be possible to drop this check in a lab via a configuration parameter or something like this? **

For context, this is the default check, i.e. the failing check for ext-container and crpd container kind. I have removed the check in the below patch:
```
diff --git a/nodes/default_node.go b/nodes/default_node.go
index 2647d154..7882a3c1 100644
--- a/nodes/default_node.go
+++ b/nodes/default_node.go
@@ -175,15 +175,6 @@ func (d *DefaultNode) DeleteNetnsSymlink() error {
 // Checks the name pattern that is common for VM-based nodes. Container-based nodes override this function to implement
 // custom name checks.
 func (d *DefaultNode) CheckInterfaceName() error {
-       // allow ethX interface names
-       // https://regex101.com/r/C3Fhr0/2
-       ifRe := regexp.MustCompile(`eth[1-9][0-9]*$`)
-       for _, e := range d.Config().Endpoints {
-               if !ifRe.MatchString(e.EndpointName) {
-                       return fmt.Errorf("%q interface name %q doesn't match the required pattern. It should be named as ethX, where X is >0", d.Cfg.ShortName, e.EndpointName)
-               }
-       }
-
        return nil
 }
```

**Comment by steiler** [comment]
Fully agree!
@hellt changed the implementation in the PR before merging. My view would also be, that in DefaultNode the CheckInterfaceName() should basically be a noop and every Kind that has specific requirements should be able to limit the interface name-space as specifically required.

Lookking up the code I noticed that the DefaultNode CheckInterfaceName function comment mentions that a common behaviour is meant to be an implemented for the VM based nodes.
My suggestion would be to come up with another "abstract" struct that sits between DefaultNode and the actual Kind.  Maybe called VRNode... that aggregates behaviour for the vr based kinds.

**Comment by steiler** [comment]
If you require a hotfix, just download this build from the PR branch:
https://github.com/srl-labs/containerlab/suites/11305858076/artifacts/580360665

**Comment by steiler** [comment]
Fixed via #1276 next release will have the fix included.
